File: OfflineFetcher/app/src/main/java/de/jbamberger/offlinefetcher/di/reddit/stream/RedditStreamSubcomponent.java
package de.jbamberger.offlinefetcher.di.reddit.stream;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;
import de.jbamberger.offlinefetcher.ui.reddit.RedditStreamFragment;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@RedditStreamScope
@Subcomponent(modules = RedditStreamModule.class)
public interface RedditStreamSubcomponent extends AndroidInjector<RedditStreamFragment> {
    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<RedditStreamFragment> {}
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/reddit/RedditStreamSubcomponent.kt
package de.jbamberger.offlineclient.ui.reddit

import dagger.Subcomponent
import dagger.android.AndroidInjector

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@RedditStreamScope
@Subcomponent(modules = [RedditStreamModule::class])
interface RedditStreamSubcomponent : AndroidInjector<RedditStreamFragment> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<RedditStreamFragment>()
}
